From ac8a962b3c3166be6891e434bc5a516ed1e39fa7 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:40 -0400
Subject: [tools] perf/lock: Introduce --list-cmds for use by scripts

Message-id: <1410181752-25631-156-git-send-email-jolsa@redhat.com>
Patchwork-id: 91802
O-Subject: [PATCH RHEL7.1 BZ1134356 155/307] perf lock: Introduce --list-cmds for use by scripts
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit a2368c3199eba493d72c6d0e5b804f908a09706c
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Fri Mar 14 23:17:53 2014 -0400
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-lock.c b/tools/perf/builtin-lock.c
index c852c7a..6148afc 100644
--- a/tools/perf/builtin-lock.c
+++ b/tools/perf/builtin-lock.c
@@ -961,8 +961,10 @@ int cmd_lock(int argc, const char **argv, const char *prefix __maybe_unused)
   "perf lock info [<options>]",
   NULL
  };
- const char * const lock_usage[] = {
-  "perf lock [<options>] {record|report|script|info}",
+ const char *const lock_subcommands[] = { "record", "report", "script",
+       "info", NULL };
+ const char *lock_usage[] = {
+  NULL,
   NULL
  };
  const char * const report_usage[] = {
@@ -976,8 +978,8 @@ int cmd_lock(int argc, const char **argv, const char *prefix __maybe_unused)
  for (i = 0; i < LOCKHASH_SIZE; i++)
   INIT_LIST_HEAD(lockhash_table + i);
 
- argc = parse_options(argc, argv, lock_options, lock_usage,
-        PARSE_OPT_STOP_AT_NON_OPTION);
+ argc = parse_options_subcommand(argc, argv, lock_options, lock_subcommands,
+     lock_usage, PARSE_OPT_STOP_AT_NON_OPTION);
  if (!argc)
   usage_with_options(lock_usage, lock_options);
 
diff --git a/tools/perf/perf-completion.sh b/tools/perf/perf-completion.sh
index f44c04b..ecedab6 100644
--- a/tools/perf/perf-completion.sh
+++ b/tools/perf/perf-completion.sh
@@ -122,7 +122,7 @@ __perf_main ()
   evts=$($cmd list --raw-dump)
   __perfcomp_colon "$evts" "$cur"
  # List subcommands for perf commands
- elif [[ $prev == @(kvm|kmem|mem) ]]; then
+ elif [[ $prev == @(kvm|kmem|mem|lock) ]]; then
   subcmds=$($cmd $prev --list-cmds)
   __perfcomp_colon "$subcmds" "$cur"
  # List long option names
-- 
1.7.1